fe/fe-core/src/main/java/com/starrocks/http/rest/transaction/TransactionWithoutChannelHandler.java

```diff
@@ -37,11 +39,23 @@ public class TransactionWithoutChannelHandler implements TransactionOperationHan
 
     private final TransactionOperationParams txnOperationParams;
 
+    // Maybe null because only begin operation need it, and it's lazily initialized in handle() method
+    @Nullable
+    private Warehouse warehouse;
```

Comment:
‌‌‌I did not find any code related to setting the warehouse in the handle() method.

Reply:
This is a newly developed feature. It is not complete for the time being. I just migrated it.